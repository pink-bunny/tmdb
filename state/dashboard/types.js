export const FETCH_TRENDING_MOVIES = 'FETCH_TRENDING_MOVIES';
export const FETCH_TRENDING_MOVIES_SUCCESS = 'FETCH_TRENDING_MOVIES_SUCCESS';
export const FETCH_TRENDING_MOVIES_ERROR = 'FETCH_TRENDING_MOVIES_ERROR';
