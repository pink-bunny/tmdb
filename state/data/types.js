export const MERGE_MOVIES_LIST = 'MERGE_MOVIES_LIST';

export const MERGE_LISTS_LIST = 'MERGE_LISTS_LIST';
