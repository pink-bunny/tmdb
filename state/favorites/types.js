export const FETCH_FAVORITES = 'FETCH_FAVORITES';
export const FETCH_FAVORITES_SUCCESS = 'FETCH_FAVORITES_SUCCESS';
export const FETCH_FAVORITES_ERROR = 'FETCH_FAVORITES_ERROR';

export const TOGGLE_TO_FAVORITES = 'TOGGLE_TO_FAVORITES';
export const TOGGLE_TO_FAVORITES_SUCCESS = 'TOGGLE_TO_FAVORITES_SUCCESS';
export const TOGGLE_TO_FAVORITES_ERROR = 'TOGGLE_TO_FAVORITES_ERROR';
