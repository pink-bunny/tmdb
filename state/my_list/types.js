export const FETCH_MY_LIST = 'FETCH_MY_LIST';
export const FETCH_MY_LIST_SUCCESS = 'FETCH_MY_LIST_SUCCESS';
export const FETCH_MY_LIST_ERROR = 'FETCH_MY_LIST_ERROR';

export const CREATE_MY_LIST = 'CREATE_MY_LIST';

export const DELETE_MY_LIST = 'DELETE_MY_LIST';
export const DELETE_MY_LIST_ERROR = 'DELETE_MY_LIST_ERROR';

export const TOGGLE_TO_LIST = 'TOGGLE_TO_LIST';
export const TOGGLE_TO_LIST_SUCCESS = 'TOGGLE_TO_LIST_SUCCESS';

export const CREATE_AND_ADD_TO_LIST = 'CREATE_AND_ADD_TO_LIST';
