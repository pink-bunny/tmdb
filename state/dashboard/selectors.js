export const isDashboardMoviesLoading = (state) => state.dashboard.loading;

export const dashboardMoviesList = (state) => {
  const refs = state.dashboard.ids;
  const list = state.data.movies;

  if (refs.length === 0) {
    return null;
  }
  return refs.map((item) => list[item]);
};

export const dashboardMoviesTotalPages = (state) => (
  state.dashboard.totalItems
);

export const dashboardMoviesCurrentPage = (state) => (
  state.dashboard.currentPage
);

export const dashboardMoviesError = (state) => state.dashboard.error;

export const dashboardMoviesSearchQuery = (state) => state.dashboard.searchQuery;
