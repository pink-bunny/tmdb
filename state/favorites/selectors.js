export const isFavoritesLoading = (state) => state.favorites.loading;

export const favoritesTotalItems = (state) => state.favorites.totalItems;

export const favoritesCurrentPage = (state) => state.favorites.currentPage;

export const favoritesError = (state) => state.favorites.error;

export const favoritesList = (state) => {
  const refs = state.favorites.ids || [];
  const list = state.data.movies || {};

  if (refs.length === 0) {
    return null;
  }
  return refs.map((item) => list[item]);
};
