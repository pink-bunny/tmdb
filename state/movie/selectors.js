export const isMovieLoading = (state) => state.movie.loading;

export const movieError = (state) => state.movie.error;

export const movieDetails = (state) => {
  const id = state.movie.id || null;
  const movie = state.data.movies || {};

  if (!id) {
    return null;
  }

  return movie[id];
};
