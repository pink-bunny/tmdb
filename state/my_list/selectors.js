export const isMyListLoading = (state) => state.myList.loading;

export const myListError = (state) => state.myList.error;

export const myListInfo = (state) => {
  const id = state.myList.movieId || null;
  const list = state.data.lists || {};

  if (!id) {
    return null;
  }
  return list[id];
};

export const deleteMyListError = (state) => state.myLists.deleteMyListError;
