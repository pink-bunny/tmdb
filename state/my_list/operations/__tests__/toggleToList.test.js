import toggleToListLogic from '../toggleToList';
import mockHttpClient from '../../../../utils/testsHelpers/mockHttpClient';
import { getSessionId } from '../../../session/selectors';
import { TOGGLE_TO_LIST } from '../../types';

describe('toggleToList logic', () => {
  const listId = 123;
  const movieId = 123;
  const action = {
    type: TOGGLE_TO_LIST,
    listId,
    movieId
  };
  const dispatch = jest.fn();
  const done = jest.fn();
  const getState = jest.fn(() => ({ session: { sessionId: '1q2w3e4r5t' } }));
  const state = getState();
  const sessionId = getSessionId(state);
  const httpClient = mockHttpClient({ method: 'post' });
  toggleToListLogic.process({ getState, httpClient, action }, dispatch, done);

  it('matches snapshot', () => {
    expect(toggleToListLogic).toMatchSnapshot();
  });

  it('called with right arguments', () => {
    const requestUrl = `/list/${listId}/add_item?session_id=${sessionId}`;
    const requestBody = {
      media_id: movieId /* eslint-disable-line camelcase */
    };
    expect(httpClient.post).toHaveBeenCalledWith(requestUrl, requestBody);
  });

  it('have been called done()', () => {
    expect(done).toHaveBeenCalled();
  });
});
