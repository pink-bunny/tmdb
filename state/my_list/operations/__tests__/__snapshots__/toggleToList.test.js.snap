// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleToList logic matches snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": false,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "TOGGLE_TO_LIST",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
